src/main/java/org/folio/consortia/service/impl/SharingSettingServiceImpl.java

```diff
@@ -96,21 +101,46 @@ public SharingSettingResponse start(UUID consortiumId, SharingSettingRequest sha
     }
   }
 
+  private UUID publishRequest(UUID consortiumId, PublicationRequest publicationRequest) {
+    if (CollectionUtils.isNotEmpty(publicationRequest.getTenants())) {
+      return publicationService.publishRequest(consortiumId, publicationRequest).getId();
+    }
+    log.info("publishRequest:: Tenant list of publishing for http method: {} is empty", publicationRequest.getMethod());
+    return null;
+  }
+
+  private void checkEqualsOfPayloadIdWithSettingId(SharingSettingRequest sharingSettingRequest) {
+    String sharingSettingId = String.valueOf(sharingSettingRequest.getSettingId());
+    String payloadId = getPayloadId(sharingSettingRequest.getPayload());
+    if (ObjectUtils.notEqual(sharingSettingId, payloadId)) {
+      throw new IllegalArgumentException("id in payload is not equal to settingId");
+    }
+  }
+
+  private String getPayloadId(Object payload) {
+    JsonNode payloadNode = objectMapper.convertValue(payload, JsonNode.class);
+    return payloadNode.get("id").asText();
+  }
+
   private PublicationRequest createPublicationRequestForSetting(SharingSettingRequest sharingSettingRequest, String httpMethod) {
     PublicationRequest publicationRequest = new PublicationRequest();
     publicationRequest.setMethod(httpMethod);
-    publicationRequest.setUrl(sharingSettingRequest.getUrl());
+    String url = sharingSettingRequest.getUrl();
+    isValidUrl(url);
```

Comment:
Can we use some standard validation tool from  apache or spring here?

Reply:
I tried Url validator from apache looks like it will check whether it starts with https also some test cases failed.

/organization/ - This URL was considered valid by UrlValidator of apache common but this URL is not work because it will throw error not found module

So created new regex pattern currently it is working good added some tests

![image](https://github.com/folio-org/mod-consortia/assets/113523904/2df7493b-8760-4996-9803-00b82cfbce15)

Reply:
In this case lets not use our custom validation, please shelve your cahnges but lets not include in PR

Reply:
As far as I know, we will get custom url like '/user/group". not like "https:/localhost:/..." so, I thought custom regex is more appropriate in that case

Reply:
Ok, reverting changes